planning/behavior_path_planner/src/scene_module/avoidance/avoidance_utils.cpp

```diff
@@ -342,4 +342,149 @@ Polygon2d createEnvelopePolygon(
   return offset_polygons.front();
 }
 
+std::vector<Point> getEdgePoints(const Polygon2d & object_polygon, const double threshold)
+{
+  if (object_polygon.outer().size() < 2) {
+    return {};
+  }
+
+  std::vector<Point> edge_points;
+  const size_t num_points = object_polygon.outer().size();
+  for (size_t i = 0; i < num_points - 1; ++i) {
+    const auto & curr_p = object_polygon.outer().at(i);
+    const auto & next_p = object_polygon.outer().at(i + 1);
+    const auto & prev_p =
+      i == 0 ? object_polygon.outer().at(num_points - 2) : object_polygon.outer().at(i - 1);
+    const Eigen::Vector2d current_to_next(next_p.x() - curr_p.x(), next_p.y() - curr_p.y());
+    const Eigen::Vector2d current_to_prev(prev_p.x() - curr_p.x(), prev_p.y() - curr_p.y());
+    const double inner_val = current_to_next.dot(current_to_prev);
+    if (std::fabs(inner_val) > threshold) {
+      continue;
+    }
+
+    const auto edge_point = tier4_autoware_utils::createPoint(curr_p.x(), curr_p.y(), 0.0);
+    edge_points.push_back(edge_point);
+  }
+  return edge_points;
+}
+
+std::vector<PolygonPoint> sortPolygonPoints(const std::vector<PolygonPoint> & points)
```

Comment:
What does this function do?
From the function name it appears to sort, but in reality it appears to return two or three points.

Reply:
This function sorts each collision point based on the lateral and longitudinal length from the current pose of the ego vehicle.

Reply:
![image](https://user-images.githubusercontent.com/8327598/206637645-484daa8c-6457-4c66-a89a-d6e5bab6c37a.png)
![image](https://user-images.githubusercontent.com/8327598/206637679-4087557a-7166-4a5e-8118-33df791f9d25.png)
From the function name, it appears that the points are sorted according to some value. But it appears to return only two or three.
It is good if @TakaHoribe and @takayuki5168 are ok.

Reply:
@satoshi-ota Could you check this.

Reply:
I check and understand this function logic. I think it looks good to me.

[imo] `fixPointsOrder`, `changePointsOrder` is better...?